src/scipp/utils/bin_edges.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Jan-Lukas Wynen
+"""
+Helpers for handling bin-edges.
+"""
+
+from typing import Set
+
+from ..core import Variable
+from ..typing import VariableLike
+
+
+def find_bin_edge_dims(ref: VariableLike, /, *, coord: Variable) -> Set[str]:
+    """
+    Returns all bin-edge dimensions of `coord` relative to the sizes of `ref`.
+
+    Parameters
+    ----------
+    ref:
+        Defines the reference dims and shape.
+    coord:
+        Defines dims to search.
+
+    Returns
+    -------
+    :
+        All labels of bin-edge dimensions.
+
+    Examples
+    --------
+
+        >>> da = sc.DataArray(sc.zeros(sizes={'a': 2, 'b': 3}))
+        >>> coord = sc.zeros(sizes={'a': 2, 'b': 4})
+        >>> sc.utils.find_bin_edge_dims(da, coord=coord)
+        {'b'}
+
+    If `ref` is a scalar, length-2 coords are assumed to be bin-edges.
+    This is required to identify bin-edges in slices.
+
+        >>> coord = sc.zeros(sizes={'a': 2})
+        >>> sc.utils.find_bin_edge_dims(da['a', 0]['b', 0], coord=coord)
+        {'a'}
+    """
+    bin_edges = set()
+    for idx, dim in enumerate(coord.dims):
+        length = coord.shape[idx]
+        if not ref.dims:
+            # Have a scalar slice.
+            # Cannot match dims, just assume length 2 attributes are bin-edge
+            if length == 2:
+                bin_edges.add(dim)
+        elif dim in ref.dims and ref.shape[ref.dims.index(dim)] + 1 == length:
+            bin_edges.add(dim)
+    return bin_edges
+
+
+def is_bin_edges(ref: VariableLike, /, *, coord: Variable) -> bool:
```

Comment:
- Missing a dimension label? What if `coord` is 2d? The return value will be unclear.
- Would this work better as a method of `Coord`, i.e., `da.coords.is_edges(name, dim)`?

Reply:
> Missing a dimension label? What if coord is 2d? The return value will be unclear.

It checks if any dim is a bin-edge. I.e. the same that the HTML output does. I don't see why we would do anything different here.

> Would this work better as a method of Coord, i.e., da.coords.is_edges(name, dim)?

Then it should also be a method of `attrs` and `meta`. And it would limit it to cases where the coord is already stored in the DataArray. What if a user wants to check before adding it because they maybe don't want to add bin-edges?

Reply:
> It checks if any dim is a bin-edge. I.e. the same that the HTML output does. I don't see why we would do anything different here.

Because this is really important for writing code that is not broken? If we have 2d data and coord we need to know *which* dim has edges. For the HTML output it is maybe not critical, or rather it should be fixed to indicate *which* dims the edges refer to?

> Then it should also be a method of attrs and meta.

Yes.

> And it would limit it to cases where the coord is already stored in the DataArray. What if a user wants to check before adding it because they maybe don't want to add bin-edges?

Seems like a very special case? Maybe we should not compromise the API for this at this point?

Reply:
Finding out which dim has edges is covered by `find_bin_edge_dims`. If we add a `dim` arg to `is_bin_edges`, it should be optional for 1d coords, I think. Otherwise, one would pretty much always write `is_bin_edges(da, da.coords['x'], da.coords['x'].dim)` (or the equivalent method).

> Maybe we should not compromise the API for this at this point?

I don't see this as a compromise. But I don't know of any concrete use case of a free function. So I am fine with changing it to a method.

Reply:
>  If we add a dim arg to is_bin_edges, it should be optional for 1d coords, I think.

Yes that makes sense. I believe we have done so in similar cases.